torch_geometric/explain/algorithm/pg_explainer.py

```diff
@@ -0,0 +1,276 @@
+import logging
+import warnings
+from typing import Optional
+
+import torch
+import torch.nn.functional as F
+import tqdm
+from torch import Tensor, nn
+from torch_scatter import scatter_mean
+
+from torch_geometric.explain import Explanation
+from torch_geometric.explain.algorithm.utils import clear_masks, set_masks
+from torch_geometric.explain.config import MaskType, ModelMode, ModelTaskLevel
+from torch_geometric.utils import get_intermediate_messagepassing_embeddings
+
+from .base import ExplainerAlgorithm
+
+
+# TODO add example.
+class PGExplainer(ExplainerAlgorithm):
+    r"""The PGExplainer model from the `"Parameterized Explainer for Graph
+    Neural Network" <https://arxiv.org/abs/2011.04573>`_ paper. It
+    uses a neural network :obj:`explainer_model` , to predict which
+    edges are crucial to a GNNs node or graph prediction.
+
+    Args:
+        epochs (int, optional): The number of epochs to train.
+            (default: :obj:`30`).
+        lr (float, optional): The learning rate to apply.
+            (default: :obj:`0.003`).
+        log (bool, optional): If set to :obj:`False`, will not log any learning
+            progress. (default: :obj:`True`)
+        **kwargs (optional): Additional hyper-parameters to override default
+            settings in
+            :attr:`~torch_geometric.explain.algorithm.PGExplainer.coeffs`.
+    """
+
+    coeffs = {
+        'edge_size': 0.05,
+        'edge_ent': 1.0,
+        'temp': [5.0, 2.0],
+        'bias': 0
```

Comment:
Should we add here the temperature and bias of L188 ?

Reply:
The function `compute_edge_mask` gets called in L123, the inputs to the function are set before that using these coefficients.